packages/web-media-box/src/dash-parser/tags/base.ts

```diff
@@ -4,6 +4,55 @@ import { ADAPTATION_SET, BASE_URL, MPD, PERIOD, REPRESENTATION } from '@/dash-pa
 import type { ManifestType, ParsedManifest } from '@/dash-parser/types/parsedManifest';
 import type { SharedState } from '@/dash-parser/types/sharedState';
 import type { PendingProcessors } from '@/dash-parser/pendingProcessors.ts';
+import type { EventScheme, ParsedManifest, UTCTimingScheme } from '../types/parsedManifest';
+import { missingRequiredAttributeWarn } from '@/dash-parser/utils/warn.ts';
+import { parseAttributes } from '@/dash-parser/parseAttributes';
+import { parseUTCTimingScheme } from '@/dash-parser/utils/parseUTCTimingScheme';
+import {
+  MPDAttributes,
+  AdaptationSetAttributes,
+  Attribute,
+  PeriodAttributes,
+  BaseURLAttributes,
+  UTCTimingAttributes,
+  EventStreamAttributes,
+  EventAttributes,
+  SegmentTemplateAttributes
+} from '@/dash-parser/consts/attributes.ts';
+
+/**
+ * 
+ * @param attributes The list of attributes taken from the tag
+ * @param expectedAttributes The expected attributes based on the DASH spec
+ * @returns A parsed and formatted list attributes.
+ */
+const formatAttributes = (
```

Comment:
This is a key change. I think it would be nice if we could have all of the expected attributes in a separate file and then have some logic to utilize those to parse and format the attributes here.

Reply:
So basically, if some attribute has a default value - we should add it to the `parsedManifest` initializer since typescript will warn us because we should such fiest as available instead of optional.

By having a separate map for attributes, we create two different sources of truth here.
I believe each Processor should keep its own list of static attribute keys, and it can also maintain its own set of required attributes...

I think we should go the same approach we already have with hls-parser:

We can ensure that required attributes are presented by utilizing `safeProcess` protected method. I suggest simply re-using the same approach, and we can probably share it between 2 parsers in the feature.

![CleanShot 2023-10-19 at 18 57 34@2x](https://github.com/dzianis-dashkevich/web-media-box/assets/98566601/8db7557d-3824-4e76-a22d-07e5f89d7c7a)
![CleanShot 2023-10-19 at 18 54 45@2x](https://github.com/dzianis-dashkevich/web-media-box/assets/98566601/d7e8cc4f-dcee-4eac-a4e4-82de25469d35)